
This is code for x96-q 1G variant with board BA301_141 v1.41

Vendor DRAM config parsed from box emmc dump by jernej's script
from https://gist.github.com/jernejsk/1957a252267e72979935f1d5236018c8:

dram_clk: 600
dram_type: 3
dram_dx_odt: 0x03030303
dram_dx_dri: 0x0e0e0e0e
dram_ca_dri: 0x00001c16
dram_odt_en: 0x00000001
dram_para1: 0x000030eb
dram_para2: 0x04000000
dram_mr0: 0x00001b50
dram_mr1: 0x00000004
dram_mr2: 0x00000010
dram_mr3: 0x00000000
dram_mr4: 0x00000000
dram_mr5: 0x00000000
dram_mr6: 0x00000000
dram_mr11: 0x00000000
dram_mr12: 0x00000000
dram_mr13: 0x00000000
dram_mr14: 0x00000000
dram_mr16: 0x00000000
dram_mr17: 0x00000000
dram_mr22: 0x00000000
dram_tpr0: 0xc0000007
dram_tpr1: 0x80000000
dram_tpr2: 0x00000100
dram_tpr3: 0x00000000
dram_tpr6: 0x33808080
dram_tpr10: 0x002f6606
dram_tpr11: 0xffffdddd
dram_tpr12: 0xfedf7657
dram_tpr13: 0x00006041

Use following settings for mainline U-Boot:
CONFIG_DRAM_CLK=600
CONFIG_DRAM_ODT_EN=y
CONFIG_DRAM_SUN50I_H616_UNKNOWN_FEATURE=y
CONFIG_DRAM_SUN50I_H616_BIT_DELAY_COMPENSATION=y
CONFIG_DRAM_SUN50I_H616_READ_CALIBRATION=y


diff --speed-large-files --no-dereference --minimal -Naur u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/dts/Makefile u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/dts/Makefile
--- u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/dts/Makefile	2023-03-20 09:48:36.031698639 +0100
+++ u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/dts/Makefile	2023-03-20 09:47:58.505033349 +0100
@@ -720,7 +720,8 @@
 	sun50i-h616-orangepi-zero2.dtb \
 	sun50i-h616-x96-mate.dtb \
 	sun50i-h616-tanix-tx6s.dtb \
-	sun50i-h616-t95.dtb
+	sun50i-h616-t95.dtb \
+	sun50i-h313-x96q-v1.dtb
 dtb-$(CONFIG_MACH_SUN50I) += \
 	sun50i-a64-amarula-relic.dtb \
 	sun50i-a64-bananapi-m64.dtb \
diff --speed-large-files --no-dereference --minimal -Naur u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/dts/sun50i-h313-x96q-v1.dts u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/dts/sun50i-h313-x96q-v1.dts
--- u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/dts/sun50i-h313-x96q-v1.dts	1970-01-01 01:00:00.000000000 +0100
+++ u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/arch/arm/dts/sun50i-h313-x96q-v1.dts	2023-03-20 09:38:58.181719749 +0100
@@ -0,0 +1,262 @@
+// SPDX-License-Identifier: (GPL-2.0+ or MIT)
+/*
+ * Copyright (C) 2020 Arm Ltd.
+ */
+
+/dts-v1/;
+
+#include "sun50i-h616.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	model = "Shenzhen Amediatech Technology Co., Ltd X96Q TV Box";
+	compatible = "amediatech,x96q", "allwinner,sun50i-h616";
+
+	aliases {
+		//ethernet0 = &emac1;
+		serial0 = &uart0;
+		mmc0 = &mmc0;
+		mmc2 = &mmc2;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	reg_vcc5v: vcc5v {
+		/* board wide 5V supply directly from the USB-C socket */
+		compatible = "regulator-fixed";
+		regulator-name = "vcc-5v";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		regulator-always-on;
+	};
+};
+
+&r_i2c {
+	status = "okay";
+
+	axp305_1: pmic@36 {
+		compatible = "x-powers,axp305", "x-powers,axp805",
+		"x-powers,axp806";
+		reg = <0x36>;
+
+		/* dummy interrupt to appease the driver for now */
+		interrupts = <GIC_SPI 103 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-controller;
+		#interrupt-cells = <1>;
+	};
+};
+
+&mmc0 {
+	vmmc-supply = <&reg_ldo2>;
+	vqmmc-supply = <&reg_ldo1>;
+	//cd-gpios = <&pio 5 6 GPIO_ACTIVE_LOW>;	/* PF6 */
+	broken-cd;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&mmc1 {
+	vmmc-supply = <&reg_ldo2>;
+	vqmmc-supply = <&reg_ldo1>;
+	broken-cd;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&mmc2 {
+	vmmc-supply = <&reg_ldo2>;
+	vqmmc-supply = <&reg_ldo1>;
+	broken-cd;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&pio {
+	vcc-pc-supply = <&reg_bldo2>;
+	vcc-pd-supply = <&reg_cldo1>;
+	vcc-pg-supply = <&reg_bldo3>;
+};
+
+&r_rsb {
+	status = "okay";
+
+	axp305: pmic@745 {
+		compatible = "x-powers,axp305", "x-powers,axp805",
+			     "x-powers,axp806";
+		reg = <0x745>;
+		//interrupt-parent = <&r_intc>;
+		interrupt-controller;
+		#interrupt-cells = <1>;
+		interrupts = <GIC_SPI 105 IRQ_TYPE_LEVEL_HIGH>;
+		x-powers,self-working-mode;
+		vina-supply = <&reg_vcc5v>;
+		vinb-supply = <&reg_vcc5v>;
+		vinc-supply = <&reg_vcc5v>;
+		vind-supply = <&reg_vcc5v>;
+		vine-supply = <&reg_vcc5v>;
+		aldoin-supply = <&reg_vcc5v>;
+		bldoin-supply = <&reg_vcc5v>;
+		cldoin-supply = <&reg_vcc5v>;
+
+		regulators {
+			reg_aldo1: aldo1 {
+				regulator-always-on;
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-name = "vcc-sys";
+			};
+
+			reg_aldo2: aldo2 {
+				regulator-always-on;
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-name = "vcc3v3-ext";
+			};
+
+			reg_aldo3: aldo3 {
+				regulator-always-on;
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-name = "vcc3v3-ext2";
+			};
+
+			reg_bldo1: bldo1 {
+				regulator-always-on;
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-name = "vcc1v8";
+			};
+
+			reg_bldo2: bldo2 {
+				regulator-always-on;
+				regulator-max-microvolt = <1800000>;
+				regulator-min-microvolt = <1800000>;
+				regulator-name = "axp806-bldo2";
+			};
+
+			reg_bldo3: bldo3 {
+				regulator-always-on;
+				regulator-max-microvolt = <1800000>;
+				regulator-min-microvolt = <1800000>;
+				regulator-name = "axp806-bldo3";
+			};
+
+			reg_bldo4: bldo4 {
+				regulator-always-on;
+				regulator-max-microvolt = <1800000>;
+				regulator-min-microvolt = <1800000>;
+				regulator-name = "axp806-bldo4";
+			};
+
+			reg_cldo1: cldo1 {
+				regulator-always-on;
+				regulator-max-microvolt = <2500000>;
+				regulator-min-microvolt = <2500000>;
+				regulator-name = "axp806-cldo1";
+			};
+
+			reg_cldo2: cldo2 {
+				regulator-always-on;
+				regulator-max-microvolt = <4200000>;
+				regulator-min-microvolt = <4200000>;
+				regulator-name = "axp806-cldo2";
+			};
+
+			reg_cldo3: cldo3 {
+				regulator-always-on;
+				regulator-max-microvolt = <3300000>;
+				regulator-min-microvolt = <3300000>;
+				regulator-name = "axp806-cldo3";
+			};
+
+			reg_dcdca: dcdca {
+				regulator-always-on;
+				regulator-min-microvolt = <810000>;
+				regulator-max-microvolt = <1520000>;
+				regulator-name = "vdd-cpu";
+			};
+
+			reg_dcdcb: dcdcb {
+				regulator-always-on;
+				regulator-min-microvolt = <1000000>;
+				regulator-max-microvolt = <2550000>;
+				regulator-name = "dcdcb";
+			};
+
+			reg_dcdcc: dcdcc {
+				regulator-always-on;
+				regulator-min-microvolt = <810000>;
+				regulator-max-microvolt = <1520000>;
+				regulator-name = "vdd-gpu-sys";
+			};
+
+			reg_dcdcd: dcdcd {
+				regulator-always-on;
+				regulator-min-microvolt = <1500000>;
+				regulator-max-microvolt = <1500000>;
+				regulator-name = "vdd-dram";
+			};
+
+			reg_dcdce: dcdce {
+				regulator-boot-on;
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-name = "vcc-eth-mmc";
+			};
+
+			reg_dcdc1: dcdc1 {
+				regulator-boot-on;
+				regulator-name = "axp1530-dcdc1";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			reg_dcdc2: dcdc2 {
+				regulator-boot-on;
+				regulator-name = "axp1530-dcdc2";
+				regulator-min-microvolt = <1540000>;
+				regulator-max-microvolt = <1540000>;
+				regulator-always-on;
+			};
+
+			reg_dcdc3: dcdc3 {
+				regulator-boot-on;
+				regulator-name = "axp1530-dcdc3";
+				regulator-min-microvolt = <1840000>;
+				regulator-max-microvolt = <1840000>;
+				regulator-always-on;
+			};
+
+			reg_ldo1: ldo1 {
+				regulator-always-on;
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-name = "vcc1v8-emmc";
+			};
+
+			reg_ldo2: ldo2 {
+				regulator-always-on;
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-name = "vcc3v3-emmc";
+			};
+
+			sw {
+				/* unused */
+			};
+		};
+	};
+};
+
+&uart0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart0_ph_pins>;
+	status = "okay";
+};
+
diff --speed-large-files --no-dereference --minimal -Naur u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/configs/x96q_v1_defconfig u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/configs/x96q_v1_defconfig
--- u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/configs/x96q_v1_defconfig	1970-01-01 01:00:00.000000000 +0100
+++ u-boot-152d1a37b4856c23976cfd0758cab654b84336cd/configs/x96q_v1_defconfig	2023-03-20 09:41:31.591714175 +0100
@@ -0,0 +1,32 @@
+CONFIG_ARM=y
+CONFIG_ARCH_SUNXI=y
+CONFIG_DEFAULT_DEVICE_TREE="sun50i-h313-x96q-v1"
+CONFIG_SPL=y
+CONFIG_DRAM_CLK=600
+CONFIG_DRAM_ODT_EN=y
+CONFIG_DRAM_SUN50I_H616_UNKNOWN_FEATURE=y
+CONFIG_DRAM_SUN50I_H616_BIT_DELAY_COMPENSATION=y
+CONFIG_DRAM_SUN50I_H616_READ_CALIBRATION=y
+CONFIG_MACH_SUN50I_H616=y
+CONFIG_R_I2C_ENABLE=y
+CONFIG_SPL_I2C=y
+CONFIG_SPL_SYS_I2C_LEGACY=y
+CONFIG_SYS_I2C_MVTWSI=y
+CONFIG_SYS_I2C_SLAVE=0x7f
+CONFIG_SYS_I2C_SPEED=100000
+#CONFIG_PHY_REALTEK=y
+#CONFIG_SUN8I_EMAC=y
+#CONFIG_I2C3_ENABLE=y
+CONFIG_LOG=y
+CONFIG_LOG_CONSOLE=y
+CONFIG_LOG_MAX_LEVEL=7
+CONFIG_SPL_LOG_MAX_LEVEL=7
+CONFIG_TPL_LOG_MAX_LEVEL=7
+CONFIG_DRAM_SUN50I_H616_DX_ODT=0x03030303
+CONFIG_DRAM_SUN50I_H616_DX_DRI=0x0e0e0e0e
+CONFIG_DRAM_SUN50I_H616_CA_DRI=0x00001c16
+CONFIG_DRAM_SUN50I_H616_ODT_EN=0x1
+CONFIG_DRAM_SUN50I_H616_TPR0=0xc0000007
+CONFIG_DRAM_SUN50I_H616_TPR10=0x002f6606
+CONFIG_DRAM_SUN50I_H616_TPR11=0xffffdddd
+CONFIG_DRAM_SUN50I_H616_TPR12=0xfedf7657
